packages/framework/src/resources/workflow/discover-action-step-factory.ts

```diff
@@ -32,7 +32,7 @@ export async function discoverActionStepFactory(
       },
       resolve: resolve as (controls: Record<string, unknown>) => Awaitable<Record<string, unknown>>,
       code: resolve.toString(),
-      options,
+      options: options as StepOptions<FromSchema<Schema>, FromSchema<Schema>>,
```

Comment:
Similar to the above validation coercion, we need a similar approach here for the generic schema context in the discovery methods.